### PR TITLE
fix(kmultiselect): react to resize [KHCP-10321]

### PR DESCRIPTION
**Pull request by portikM**
# Summary

Addresses: https://konghq.atlassian.net/browse/KHCP-10321

Fix KMultiselect behaviour on window resize

**Functional testing instructions:**
Compare component behaviour in `alpha` branch [sandbox](https://kongponents-sandbox.netlify.app/#/multiselect) (incorrect) to this branch [sandbox](https://deploy-preview-2039--kongponents-sandbox.netlify.app/#/multiselect). 

## PR Checklist

* [ ] **Conventional Commits** all commits follow the conventional commit standards [outlined in the main README](https://github.com/Kong/kongponents#committing-changes).
* [ ] **Tests coverage:** test coverage was added for new features and bug fixes
* [ ] **Docs:** includes a technically accurate README

### CODE REVIEW

**Comment by portikM** [comment]
Added _do not merge_ label to merge it in time for next adoption phase (phase 9)

**Review by mihai-peteu — APPROVED** [review]

File: src/components/KMultiselect/KMultiselect.vue [hunk 1]
```diff
@@ -1015,12 +1016,17 @@ watch(() => props.items, (newValue, oldValue) => {
 }, { deep: true, immediate: true })
 
 const numericWidth = ref<number>(300)
-const setNumericWidth = (): void => {
+const setNumericWidth = async (): Promise<void> => {
+  numericWidth.value = 300
```
Comment (mptap) [inline src/components/KMultiselect/KMultiselect.vue:1020]: Is this intentional as the default value is already initialized to 300?

Comment (portikM) [reply src/components/KMultiselect/KMultiselect.vue:1020]: Yes it is - it's needed to reset to default value before we assign the value of element width on line 1022

**Review by mptap — APPROVED** [review]
👍

**Pull request merged**